src/Perl6/Metamodel/Dispatchers.nqp

```diff
@@ -191,10 +192,19 @@ class Perl6::Metamodel::WrapDispatcher is Perl6::Metamodel::BaseDispatcher {
         my $next_dispatcher := nqp::existskey($lexpad, '$*NEXT-DISPATCHER')
                                 ?? nqp::atkey($lexpad, '$*NEXT-DISPATCHER')
                                 !! nqp::null();
-        self.new(:@candidates, :idx(1), :$next_dispatcher)
+        my $stashed_dispatcher := nqp::existskey($lexpad, '$*DISPATCHER')
+                                    ?? nqp::atkey($lexpad, '$*DISPATCHER')
+                                    !! nqp::null();
```

Comment:
I thought nqp::atkey already returns nqp::null for a non-existent key?

Reply:
It does for normal NQP hashes, afaik.  Not for stashes, for some reason.